File: 32-practical-oojs/teacher.rb
class Teacher
    attr_accessor :first_name, :last_name
  
    @@all = []
  
    def initialize(first_name, last_name)
      @first_name = first_name
      @last_name = last_name
      @@all << self
    end
  
    def introduce
      puts "My name is #{self.first_name}"
    end
  
    def self.all
      @@all
    end
  
    def self.find_by_name(name)
      self.all.find do |teacher_instance|
        teacher_instance.first_name == name
      end
    end
  
end
  
# create two instances
  p1 = Teacher.new("Kent C.", "Strait")
  p2 = Teacher.new("Anita", "Plummer")
  
  p1.introduce
  
  
  Teacher.find_by_name("Anita")

File: 32-practical-oojs/Teacher.js
class Person {

    constructor(favoriteColor) {
        this.health = 100
        this.favoriteColor = favoriteColor
    }

    nap() {
        console.log('...zzzzzzz 😴')
    }
}



class Teacher extends Person {

    static all = []
    constructor(firstName, lastName) {
        super("blue") // will invoke Person's constructor
        this.firstName = firstName
        this.lastName = lastName
        Teacher.all.push(this)
    }

    introduce() {
        console.log(`Hi, my name is ${this.firstName}`)
    }

    // overriding Person's nap method
    nap() {
        console.log('I want to take a nap!!!')
    }

    static findTeacherByName(name) {
        return Teacher.all.find(teacher => teacher.firstName === name)
    }

}

const raffy = new Teacher("Rafferty", "Rios")
const michelle = new Teacher("Michelle", "Rios")
